### ISSUE TITLE
Just a question, is the below example a known SQLite issue?

### ISSUE THREAD

**Issue by JohnMount** [r-dbi/RSQLite#244]
Just a question, is the below example a known SQLite issue?  We should have column `r1` equal to column `r2`, but clearly the `random()` command is re-running on each use (not landing values in the sub-query).

``` r
my_db <- DBI::dbConnect(RSQLite::SQLite(), ":memory:")
q <- "SELECT r AS r1, r AS r2 FROM ( SELECT random() AS r FROM ( SELECT * from ( VALUES(1),(2) ) ) a ) b"
DBI::dbGetQuery(my_db, q)
#>                    r1                  r2
#> 1 9122036933615688494 5898623711603982044
#> 2 5851272419602927035  -22037614439271276
```

**Comment by krlmlr** [comment]
On my system I see:

```
$ sqlite3 --version
3.19.3 2017-06-08 14:26:16 0ee482a1e0eae22e08edc8978c9733a96603d4509645f348ebf55b579e89636b
$ echo 'SELECT r AS r1, r AS r2 FROM ( SELECT random() AS r FROM ( SELECT * from ( VALUES(1),(2) ) ) a ) b' | sqlite3
-1735496274670600037|-5897357643573398721
5701578188233199541|5793460307658272482
```

May be worth to report to the sqlite3 mailing list. The currently released version is 3.21.0, the last mention of `random()` in the [release notes](http://sqlite.org/changes.html) was in 2014.

**Comment by JohnMount** [comment]
Thanks, I did send to the sqlite3 mailing list.  I haven't heard back yet.
